micropython/build.sh

```diff
@@ -21,24 +24,55 @@ mkdir -p "$PREFIX_UPYTH_BUILD" "$PREFIX_UPYTH_MARKERS"
 
 
 #
-# Apply patches
+# Apply patches and copy files
 #
 # FIXME: prefix patch files with numbers to ensure correct order
-for patchfile in "$PREFIX_UPYTH_PATCHSET"/*.patch; do
+for patchfile in "$PREFIX_UPYTH_CONFIG"/patches/*.patch; do
 	if [ ! -f "$PREFIX_UPYTH_MARKERS/$(basename "$patchfile").applied" ]; then
 		echo "applying patch: $patchfile"
 		patch -d "$PREFIX_UPYTH_SRC" -p1 < "$patchfile"   # FIXME: should be -p1
 		touch "$PREFIX_UPYTH_MARKERS/$(basename "$patchfile").applied"
 	fi
 done
+cp -a "${PREFIX_UPYTH_CONFIG}/files/001_mpconfigport.mk" "${PREFIX_UPYTH_SRC}/ports/unix/mpconfigport.mk" && echo "Copied mpconfigport.mk!"
+
+
+#
+# Micropython internal use stack/heap size (not actual application stack/heap)
+# May be overwritten in _targets/build.project.*
+#
+if [[ ! $UPYTH_STACKSZ ]]; then
+	UPYTH_STACKSZ="4096"
+fi
+if [[ ! $UPYTH_HEAPSZ ]]; then
+	UPYTH_HEAPSZ="16384"
+fi
+
+
+#
+# Architecture specific flags/values set
+#
+if [ "${TARGET_FAMILY}" = "armv7m7" ]
+then
+	export LDFLAGS_EXTRA=" -Wl,-q ${CFLAGS}"
```

Comment:
maybe we should set the variables inside IF and then just export it once - outside of IF

Other stuff:
 - when linking by GCC we should always include `CFLAGS` while linking (so this is not `armv7m7`-specific)
 - I think we're missing the correct `-z max-page-size` settings (this at least prevented `micropython` to run on `imx6ull`
 - it probably would be wise to unexport `LDFAGS` and `CFLAGS` and remove patches changing `LDFAGS += ...` to `LDFLAGS = ...`
 -  I would go for exporting necessary `STRIPFLAGS` from `phoenix-rtos-build` in the near future (the same as `CFLAGS`/`LDFLAGS` currently) to be able to remove this if-else altogether.

I tried to come up with some generic way of changing `ld` params syntax to `gcc` params syntax, came up not with generic one, but I think it fits our requirements (so we could remove most of this if-else):
```bash
LDFLAGS_EXTRA="$(echo "$LDFALGS" | sed "s/-z \([a-z=0-9\-]*\)/ -Wl,z,\1/g; s/-q/-Wl,q/g")"
```

but it probably needs some more testing.

Reply:
- moved `CFLAGS_EXTRA` export outside of _if-else_
 - for some reason your _sed_ command did not work _plug and play_ and I got lost in trying to understand/fix it, so i went for two sed approach
 - one `STRIPFLAGS` export

@nalajcie 
 - It is good idea to just compose new `CFLAGS` and `LDFLAGS` in `build.sh` of upyth instead of patching source code and then provide additional arguments. Should I temporarily save `CFLAGS` and `LDFLAGS` as stated below:
```
CFLAGS_TMP=${CLAGS}
unset CFLAGS
make (...)
export CFLAGS="${CFLAGS_TMP} 
```
and the same for LDFLAGS?

Reply:
Just to respond Your questions:
 - yes, we should minimize the patches by setting variables in `build.sh`
 - no need to restore the environment, every `build.sh` is executed as a subprocess of phoenix-rtos-ports root `build.sh` - so it doesn't affect parent process in any way